lib/testpool/fork-test-pool.js

```diff
@@ -0,0 +1,118 @@
+const Bluebird = require('bluebird');
+const getPort = require('get-port');
+const debug = require('debug')('ava:pool');
+const {Fork} = require('../fork');
+const TestPool = require('./test-pool');
+
+// Break up files into chunks and
+// distribute to forks to run as SingleProcess
+class ForkTestPool extends TestPool {
+	bail() {
+		debug('bailing forks');
+		this.forks.map(fork => fork.exit());
+	}
+
+	run() {
+		const {concurrency, options, precompilation, runStatus} = this;
+		const {failFast} = this.initOptions;
+		const numEach = Math.floor(this.files.length / concurrency);
+		const files = Object.assign([], this.files);
+		const fileChunks = [];
+
+		for (let i = 0; i < concurrency; i++) {
+			let size = numEach;
+			if (i === concurrency - 1) {
+				size = files.length;
+			}
+
+			if (size) {
+				fileChunks.push(files.splice(0, size));
+			}
+		}
+
+		if (precompilation) {
+			options.cacheDir = precompilation.cacheDir;
+			options.precompiled = precompilation.map;
+		} else {
+			options.precompiled = {};
+		}
+
+		options.concurrency = concurrency;
+		this.forks = [];
+
+		return Bluebird.map(fileChunks, curFiles => {
+			return this._computeForkExecArgv().then(execArgv => {
+				this.forks.push(
+					new Fork(curFiles, runStatus, this.env, this.initOptions, execArgv)
```

Comment:
This… gives a single `Fork` the responsibility to run multiple test files? I can't quite tell how `Fork` implements this but it also seems a little odd. You could end up with a slow fork that still has two pending files while other slots are idle.